src/UnitSystem.cc

```diff
@@ -28,14 +28,44 @@
 
 namespace lsst { namespace meas { namespace modelfit {
 
-UnitSystem::UnitSystem(afw::coord::Coord const & position, Scalar mag) {
+UnitSystem::UnitSystem(afw::coord::Coord const & position,
+                       std::shared_ptr<const lsst::afw::image::Calib> calibIn,
+                       double flux) {
     double cdelt = (1.0*lsst::afw::geom::arcseconds).asDegrees();
     wcs = afw::image::makeWcs(position, afw::geom::Point2D(0.0, 0.0), cdelt, 0.0, 0.0, cdelt);
+    calibIn = validateCalib(calibIn);
+    Scalar mag = calibIn->getMagnitude(flux);
     PTR(afw::image::Calib) calib_ = std::make_shared<afw::image::Calib>();
     calib_->setFluxMag0(std::pow(10.0, mag/2.5));
     calib = calib_;
 }
 
+UnitSystem::UnitSystem(afw::coord::Coord const & position, Scalar mag) {
+        double cdelt = (1.0*lsst::afw::geom::arcseconds).asDegrees();
+        wcs = afw::image::makeWcs(position, afw::geom::Point2D(0.0, 0.0), cdelt, 0.0, 0.0, cdelt);
+        PTR(afw::image::Calib) calib_ = std::make_shared<afw::image::Calib>();
+        calib_->setFluxMag0(std::pow(10.0, mag/2.5));
+        calib = calib_;
+    }
+
+std::shared_ptr<const lsst::afw::image::Calib> UnitSystem::validateCalib(
+        std::shared_ptr<const lsst::afw::image::Calib> calib_) {
+        if (calib_->getFluxMag0().first == 0.0){
+            return getDefaultCalib();
+        } else{
+            return calib_;
+        }
+    }
+
+std::shared_ptr<const lsst::afw::image::Calib> UnitSystem::getDefaultCalib() {
+        // Create a default calib object with a zero-point set to magnitude 27
+        static std::shared_ptr<const lsst::afw::image::Calib> tmp =
+                                                            std::make_shared<const lsst::afw::image::Calib>(
+                                                            std::pow(10.0, 27.0/2.5)
+                                                            );
+        return tmp;
+    }
+
```

Comment:
Indentation in the above three functions is off too - looks like everything but the first line of each function is one tab stop too much.